src/MSBuild/ValidateMSBuildPackageDependencyVersions.cs

```diff
@@ -0,0 +1,87 @@
+        [Required]
+        public string AppConfig { get; set; }
+        [Required]
+        public string AssemblyPath { get; set; }
+
+        public override bool Execute()
+        {
+            XmlDocument doc = new XmlDocument();
+            doc.Load(AppConfig);
+            foreach (var topLevelElement in doc.ChildNodes)
+            {
+                if (topLevelElement is XmlElement topLevelXmlElement && topLevelXmlElement.Name.Equals("configuration", StringComparison.OrdinalIgnoreCase))
+                {
+                    foreach (var configurationElement in topLevelXmlElement.ChildNodes)
+                    {
+                        if (configurationElement is XmlElement configurationXmlElement && configurationXmlElement.Name.Equals("runtime", StringComparison.OrdinalIgnoreCase))
+                        {
+                            foreach (var runtimeElement in configurationXmlElement.ChildNodes)
+                            {
+                                if (runtimeElement is XmlElement runtimeXmlElement && runtimeXmlElement.Name.Equals("assemblyBinding", StringComparison.OrdinalIgnoreCase))
+                                {
+                                    foreach (var assemblyBindingElement in runtimeXmlElement.ChildNodes)
+                                    {
+                                        if (assemblyBindingElement is XmlElement assemblyBindingXmlElement && assemblyBindingXmlElement.Name.Equals("dependentAssembly", StringComparison.OrdinalIgnoreCase))
+                                        {
+                                            string name = string.Empty;
+                                            string version = string.Empty;
+                                            foreach (var dependentAssembly in assemblyBindingXmlElement.ChildNodes)
+                                            {
+                                                if (dependentAssembly is XmlElement dependentAssemblyXmlElement)
+                                                {
+                                                    if (dependentAssemblyXmlElement.Name.Equals("assemblyIdentity", StringComparison.OrdinalIgnoreCase))
+                                                    {
+                                                        foreach (var assemblyIdentityAttribute in dependentAssemblyXmlElement.Attributes)
+                                                        {
+                                                            if (assemblyIdentityAttribute is XmlAttribute assemblyIdentityAttributeXmlElement && assemblyIdentityAttributeXmlElement.Name.Equals("name", StringComparison.OrdinalIgnoreCase))
+                                                            {
+                                                                name = assemblyIdentityAttributeXmlElement.Value;
+                                                            }
+                                                        }
```

Comment:
You can simplify this as

```suggestion
                                                        name = dependentAssemblyXmlElement.GetAttribute("name");
```